### PR TITLE
Toggle Button

### PR DESCRIPTION
**Pull request by yasabere**
-Implemented toggle button component
-Added Toggle button examples
-Added tests

### CODE REVIEW

File: packages/react-ui-core/src/ToggleButton/__tests__/ToggleButton-test.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    const onClick = jest.fn()
+
+    const wrapper = shallow(<ToggleButton onClick={onClick} {...props} />)
+    wrapper.find('[data-tid="toggleButton"]').simulate('click')
+    expect(onClick.mock.calls[0][0]).toEqual(true)
+    wrapper.find('[data-tid="toggleButton"]').simulate('click')
+    expect(onClick.mock.calls[1][0]).toEqual(false)
+    wrapper.find('[data-tid="toggleButton"]').simulate('click')
+    expect(onClick.mock.calls[2][0]).toEqual(true)
+    console.log(onClick.mock.calls)
```
Comment (tadjohnston) [inline packages/react-ui-core/src/ToggleButton/__tests__/ToggleButton-test.js]: Whoops!

File: stories/theme/core/ToggleButton.css [hunk 2]
```diff
@@ -0,0 +1,16 @@
+.ToggleButton {
+  text-align: center;
+  color: darkgrey;
+}
+
+.ToggleButton-On svg{
```
Comment (tadjohnston) [inline stories/theme/core/ToggleButton.css]: Space needed.

File: stories/theme/core/ToggleButton.css [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  fill: #4A8FE2;
+  stroke: #4A8FE2;
+}
+
+.ToggleButton-Off svg{
```
Comment (tadjohnston) [inline stories/theme/core/ToggleButton.css]: Space needed.

File: stories/core/ToggleButton.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    )}
+    onClick={value => {
+      action('The button is ')(`${(value) ? 'toggled' : 'untoggled'}`)
+    }}
+  >
+    <span className="DefaultToggleButt" >Default Title</span>
+  </ToggleButton>
+)
+
+export const ToggleSVG = (
```
Comment (tadjohnston) [inline stories/core/ToggleButton.js]: This isn't really any different than the first story.  How about we just use this one instead?

File: packages/react-ui-core/src/ToggleButton/__tests__/ToggleButton-test.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+describe('ToggleButton', () => {
+  it('renders', () => {
+    const props = {
+      theme,
+      className: theme.ToggleButton,
+      children: <a href="#">Toggle</a>,
+    }
+
+    const wrapper = shallow(<ToggleButton {...props} />)
+    expect(wrapper.find('[data-tid="toggleButton"]').exists()).toBeTruthy()
```
Comment (tadjohnston) [inline packages/react-ui-core/src/ToggleButton/__tests__/ToggleButton-test.js]: You seem to be doing: 
```
wrapper.find('[data-tid="toggleButton"]')
```
all over this file.  I think you can just use the wrapper without the find which should give you the root.

File: stories/core/ToggleButton.js (hunk 4 above)
Comment (yasabere) [reply stories/core/ToggleButton.js]: Removed

File: stories/theme/core/ToggleButton.css (hunk 2 above)
Comment (yasabere) [reply stories/theme/core/ToggleButton.css]: Fixed

File: stories/theme/core/ToggleButton.css (hunk 3 above)
Comment (yasabere) [reply stories/theme/core/ToggleButton.css]: Fixed

File: packages/react-ui-core/src/ToggleButton/__tests__/ToggleButton-test.js (hunk 1 above)
Comment (yasabere) [reply packages/react-ui-core/src/ToggleButton/__tests__/ToggleButton-test.js]: Fixed

File: packages/react-ui-core/src/ToggleButton/__tests__/ToggleButton-test.js (hunk 5 above)
Comment (yasabere) [reply packages/react-ui-core/src/ToggleButton/__tests__/ToggleButton-test.js]: Fixed

File: packages/react-ui-core/src/ToggleButton/__tests__/ToggleButton-test.js [hunk 6]
```diff
@@ -0,0 +1,72 @@
+import React from 'react'
+import { shallow } from 'enzyme'
+import ThemedToggleButton from '../ToggleButton'
+import theme from './mocks/theme'
+
+const ToggleButton = ThemedToggleButton.WrappedComponent
+
+describe('ToggleButton', () => {
+  it('renders', () => {
```
Comment (tadjohnston) [inline packages/react-ui-core/src/ToggleButton/__tests__/ToggleButton-test.js]: `it('renders')` really doesn't tell me anything. They all render or fail.  How about testing for one specific thing here like children or the base class?

File: packages/react-ui-core/src/ToggleButton/index.js [hunk 7]
```diff
@@ -0,0 +1 @@
+export { default as ToggleButton } from './ToggleButton'
```
Comment (tadjohnston) [inline packages/react-ui-core/src/ToggleButton/index.js]: Please export like everything else is exported:
```
export { default } from './ToggleButton'
export { default as ToggleButton } from './ToggleButton'
```

Also, this probably belongs in `react-ui-core/src/Button` directory.

File: packages/react-ui-core/src/ToggleButton/ToggleButton.js [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+      className,
+      theme,
+      children,
+      onClick,
+      value,
+      ...props
+    } = this.props
+
+    return (
+      <button
```
Comment (tadjohnston) [inline packages/react-ui-core/src/ToggleButton/ToggleButton.js]: You should be using all react-ui-core components when possible i.e. `<Button/>`

File: packages/react-ui-core/src/ToggleButton/__tests__/ToggleButton-test.js (hunk 6 above)
Comment (yasabere) [reply packages/react-ui-core/src/ToggleButton/__tests__/ToggleButton-test.js]: Fixed

File: packages/react-ui-core/src/ToggleButton/index.js (hunk 7 above)
Comment (yasabere) [reply packages/react-ui-core/src/ToggleButton/index.js]: Fixed

File: packages/react-ui-core/src/ToggleButton/ToggleButton.js (hunk 8 above)
Comment (yasabere) [reply packages/react-ui-core/src/ToggleButton/ToggleButton.js]: Changed to Button

File: packages/react-ui-core/src/Button/ToggleButton.js [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+
+    return (
+      <Button
+        onClick={this.toggle}
+        className={classnames(
+          theme.ToggleButton,
+          className,
+          theme[this.state.value ? 'ToggleButton-On' : 'ToggleButton-Off'],
+        )}
+        data-tid="toggleButton"
```
Comment (tadjohnston) [inline packages/react-ui-core/src/Button/ToggleButton.js]: This should be `toggle-button`, not `toggleButton`

Comment (yasabere) [reply packages/react-ui-core/src/Button/ToggleButton.js]: Fixed

File: packages/react-ui-core/src/Button/index.js [hunk 10]
```diff
@@ -1,2 +1,4 @@
 export { default } from './Button'
 export { default as Button } from './Button'
+
```
Comment (bethstevic) [inline packages/react-ui-core/src/Button/index.js]: Can you delete this extra line?

File: packages/react-ui-core/src/index.js [hunk 11]
```diff
@@ -87,3 +87,8 @@ export {
 export {
   Highlighter,
 } from './Highlighter'
+
+export {
+  ToggleButton,
+} from './ToggleButton'
+
```
Comment (bethstevic) [inline packages/react-ui-core/src/index.js]: Wouldn't you export ToggleButton from Button?

**Comment by bethstevic** [comment]
Other than those minor nit-picks looks good 👍

File: packages/react-ui-core/src/Button/ToggleButton.js [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+  render() {
+    const {
+      className,
+      theme,
+      children,
+      onClick,
+      value,
+      ...props
+    } = this.props
+
```
Comment (nray621) [inline packages/react-ui-core/src/Button/ToggleButton.js]: Maybe we should do a `componentWillReceiveProps` here to check against `newProps` to make sure the state gets reset when a new value gets passed in?

File: packages/react-ui-core/src/index.js (hunk 11 above)
Comment (yasabere) [reply packages/react-ui-core/src/index.js]: This was a suggestion from tad
"Please export like everything else is exported:

export { default } from './ToggleButton'
export { default as ToggleButton } from './ToggleButton'
Also, this probably belongs in react-ui-core/src/Button directory."

File: packages/react-ui-core/src/Button/ToggleButton.js (hunk 12 above)
Comment (yasabere) [reply packages/react-ui-core/src/Button/ToggleButton.js]: Fixed

Comment (nray621) [reply packages/react-ui-core/src/Button/ToggleButton.js]: 👍

**Comment by nray621** [comment]
LGTM 👍

File: packages/react-ui-core/src/Button/ToggleButton.js [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+    return (
+      <Button
+        onClick={this.toggle}
+        className={classnames(
+          theme.ToggleButton,
+          className,
+          theme[this.state.value ? 'ToggleButton-On' : 'ToggleButton-Off'],
```
[the remark this answers is not in the document]
Comment (yasabere) [reply packages/react-ui-core/src/Button/ToggleButton.js]: Changed

**Pull request merged**